### PR TITLE
feat: add category in the url

### LINKED ISSUE #1620: [OTHER] add category name to the url
### What would you like to share or ask?

Hi, I want to add the category name in the URL when the dropdown button gets clicked. For example, when user would visit https://linkshub.dev/languages then it should redirect to the **open source section** and open the open-source dropdown by default

Currently: 
image

Should be like this: 
image

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by rupali-codes**
## Fixes Issue

Closes #1620 

## Changes proposed

* Added category name in the URL: whenever user would visit any specific category it would auto-expand the sidebar
* Created a default page for category: whenever user would visit the category without any subcategory it would render a default page along with category and its resources

## Screenshots

image

## Note to reviewers

Please review carefully and let me know if any changes required!

### CODE REVIEW

**Review by CBID2 — APPROVED** [review]
LGTM

**Review by Anmol-Baranwal — DISMISSED** [review]
I think the heading and sub-heading with underline should be the name of the category.

image

Like, will you prefer `Internet-Of-Things` OR `Internet Of Things`

Anyway, up to you :)

I have approved the PR.

**Comment by Anmol-Baranwal** [comment]
> Few more observations:
> 
> 1. Isn't it redundant to render the same information twice? How about rendering all the links (cards) which fall under that `category` instead?

@abhijithmuthyala 
Hey hey hey!

You are doing great; thanks for your valuable observations. They are indeed helpful, except for the first point. Many companies provide alternatives to achieve the same goal. While it may not be correct to display the data in the same way, in this case, I think it is perfectly fine. For example, just go to Amazon and see how many options they offer to achieve the same goal.

**Comment by rupali-codes** [comment]
> I think the heading and sub-heading with underline should be the name of the category.
> 
> image
> 
> Like, will you prefer `Internet-Of-Things` OR `Internet Of Things`
> 
> Anyway, up to you :)
> 
> I have approved the PR.

i think, removing hyphens would be better

**Comment by abhijithmuthyala** [comment]
> You are doing great; thanks for your valuable observations. 

@Anmol-Baranwal Thanks!

> While it may not be correct to display the data in the same way, in this case, I think it is perfectly fine. For example, just go to Amazon and see how many options they offer to achieve the same goal.

I think one advantage is that on mobile/tab screens, the links are visible right away without having to first open the sidebar. On wider screens though, the repetition is right on the face, but that's probably just me.

**Comment by rupali-codes** [comment]
@abhijithmuthyala & @Anmol-Baranwal 👀

**Comment by Anmol-Baranwal** [comment]
> @abhijithmuthyala & @Anmol-Baranwal 👀

Why did you tag me?

The hyphens seems to be there. Is there an issue?

File: pages/[category]/index.tsx [hunk 1]
```diff
@@ -19,30 +19,40 @@ const CategoryPage = () => {
...
+                </Link>
+              ))}
+            </div>
+          </>
+        ) : (
+          <div className='text-center'>
+            <p className="text-theme-secondary dark:text-gray-text w-fit mx-auto text-center">
+              <span className='md:text-7xl text-xl uppercase font-bold '>404|</span> No resources found.
+            </p>
+          </div>
```
Comment (abhijithmuthyala) [inline pages/[category]/index.tsx:54]: The request's response status itself should be 404. Here the request is ok (200), but we are rendering a 404 message. The correct way to go about it is to implement `getStaticPaths` with `fallback: false`. That way request to a route that is not included in `getStaticPaths` will 404 and by default nextjs will render the message. 

You can check the `[...subcategory].tsx` file for reference.

Comment (rupali-codes) [reply pages/[category]/index.tsx:54]: i guess its okay..

**Comment by rupali-codes** [comment]
> > @abhijithmuthyala & @Anmol-Baranwal 👀
> 
> Why did you tag me?
> 
> The hyphens seems to be there (not removed). Is there an issue?

nh, just wanted you to review

**Review by Anmol-Baranwal — APPROVED** [review]
LGTM  :)

**Review by CBID2 — APPROVED** [review]
LGTM

File: pages/[category]/index.tsx (hunk 1 above)
Comment (abhijithmuthyala) [reply pages/[category]/index.tsx:54]: It's a bug.

**Pull request merged**